### PR TITLE
makeDarwinImage: add option to not run the installer a second time

### PR DESCRIPTION
**Pull request by Luflosi**
Add an option `createDiskImageIfNotExists` which will prevent running the installer again and creating new copies of macOS in the Nix store whenever a dependency of NixThePlanet changes in Nixpkgs.

I override the Nixpkgs input on all my flake inputs so I don't have a dozen different copies of it. This causes the macOS installer to run almost every time I update, creating another version of the macOS base image. I don't need these extra copies, since I just keep the original install. Even when running the Nix garbage collector, I always have at least two macOS base images in my Nix store. One of them is referred to by the run script and the other one is referred to by the symlink in `/var/lib/nixtheplanet-macos-ventura`. This new option will allow me to never run the installer again and only have one version of macOS in my Nix store. If I ever delete the two files in `/var/lib/nixtheplanet-macos-ventura`, I can just enable this option again for a short amount of time until I have a new base image.

### CODE REVIEW

File: makeDarwinImage/run.nix [hunk 1]
```diff
@@ -45,9 +46,18 @@ writeShellScriptBin "run-macOS.sh" ''
     "$@"
   )
 
+  if [ ! -L macos-ventura-base-image.qcow2 ]; then
+  ${if createDiskImageIfNotExists then ''
+    echo 'The disk image macos-ventura-base-image.qcow2 was not found, creating it:'
+    nix-store --realise ${diskImage} --add-root macos-ventura-base-image.qcow2
+  '' else ''
+    echo 'The disk image macos-ventura-base-image.qcow2 was not found but we don'"'"'t have the ability to create it. Please enable the services.macos-ventura.createDiskImageIfNotExists option and try again.'
```
Comment (MatthewCroughan) [inline makeDarwinImage/run.nix:54]: `don'"'"'t`

Can you just say "do not" instead?

Comment (Luflosi) [reply makeDarwinImage/run.nix]: Sure.

**Comment by Luflosi** [comment]
Could you please take another look?

**Comment by MatthewCroughan** [comment]
Sure, running this in CI now, will merge when it passes.

**Comment by Luflosi** [comment]
Awesome, thanks!

**Pull request merged**